sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,651 @@
+
+The base pointer determines the starting address of the matrix to be loaded/stored. `layout` determines whether the data are being read/written in a row (`row_major`), column major (`column_major`) fashion, or if the data has already been transformed into VNNI format (`packed`). `stride` describes the number of elements between consecutive rows for row major and packed layout,  columns for column major layout. 
+
+Note that for getting maximum performance on Intel AMX and DPAS, prepacking data in the memory is necessary. If users did not specify the packed layouts, transforms done by the implementation will be slow due to extra scatter/gather operations. Hence, we expose the `packed` layout to the user to specify that A or B have already been VNNIed. The packed or VNNI layout is introduced in `VNNI layout` section below.
+	
+IMPORTANT: In the current AMX and DPAS implementation, the layout in the load of matrix B must be `packed_b` or `row_major`. Automatic VNNI transform is supported. The layout in the load of matrices A and C must be `row_major`, and the layout in the store of matrix C must also be `row_major`.
+
+Since the matrix functions are group operations (as defined in Section 4.17.3 of the SYCL specification), the matrix API has to be accessed by all the work-items in the group in a convergent control flow. The `Group` template argument can be a work-group or a subgroup. These functions will be called once by each work item in the group.
+
+To be aligned with the SYCL 2020 group algorithms, an additional group argument is added to the matrix operations to designate that these functions are collective operations. The {dpcpp} syntax is the following: 
+
+IMPORTANT: In the current implementation, only the subgroup scope is supported.  
+
+#### Load 
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_use U, matrix_layout L,
+          access::address_space Space>
+  void joint_matrix_load(Group sg, joint_matrix<T, NumRows, NumCols, U, L, Group> &res,
+		    multi_ptr<T, Space> src, size_t stride, matrix_layout memL);
+}
+```
+This function loads data from memory to the 2d tiles/registers of Intel AMX/DPAS.
+
+
+#### Store 
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+  template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_use U, matrix_layout L,
+          access::address_space Space>	  
+  void joint_matrix_store(Group sg, joint_matrix<T, NumRows, NumCols, U, L, Group> &res,
+		     multi_ptr<T, Space> src, size_t stride, matrix_layout memL);
+}
+```
+This function stores the data from the 2d tiles back to memory.
+
+#### Multiply and Add
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+  template <typename Group, typename Ta, typename Tb, typename Tc,
```

Comment:
Does Intel hardware support `Ta != Tb`? I'm surprised if they would ever be different types?

Also in the CUDA + HIP AMD backends it is not necessary to provide the matrix_layout (doc should be updated to "layout" at some point) template parameter to `joint_matrix_mad`: it looks like it is always necessary for intel hardware? if so I can add it to the CUDA implementation so that the interfaces match.